src/reactor/mod.rs

```diff
@@ -358,11 +368,24 @@ impl fmt::Debug for Reactor {
 impl Handle {
     /// Returns a handle to the current reactor.
     pub fn current() -> Handle {
-        Handle::default()
+        Handle::try_current()
+            .unwrap_or(Handle { inner: Weak::new() })
+    }
+
+    /// Try to get a handle to the current reactor.
+    ///
+    /// Returns `Err` if no handle is found.
+    pub(crate) fn try_current() -> io::Result<Handle> {
```

Comment:
It might be useful to expose this as a public API, but for now it is kept private.